gulpfile.js

```diff
@@ -234,8 +234,21 @@ function endBuildStep(stepName, targetName, startTime) {
  * @return {!Promise}
  */
 function buildExtensions(options) {
-  var results = [];
-  for (var key in extensions) {
+  const results = [];
+  let extensionToBuild = null;
+  if (argv.extension) {
+    let arg = argv.extension;
+    arg = arg.split(',');
+    extensionToBuild = Array.isArray(arg) ? arg : [arg];
+  }
+  for (const key in extensions) {
+    const extensionName = key.substring(0, key.length - 4);
+    if (argv.extension && extensionToBuild) {
+      if (extensionToBuild.indexOf(extensionName) < 0) {
+        // Skip this extension;
+        continue;
+      }
+    }
```

Comment:
Here's a simpler way to rewrite this function:

```
function buildExtensions(options) {
  var extensionsToBuild = [];
  if (!!argv.extensions) {
    extensionsToBuild = argv.extensions.split(',');
  }
  var results = [];
  for (var key in extensions) {
    var e = extensions[key];
    var o = Object.assign({}, options);
    o = Object.assign(o, e);
    if (extensionsToBuild.length > 0 &&
        extensionsToBuild.indexOf(e.name) == -1) {
      continue;
    }
    results.push(buildExtension(e.name, e.version, e.hasCss, o, e.extraGlobs));
  }
  return Promise.all(results);
}
```

Reply:
For logging, add this code snippet to `performBuild()` after the call to `printConfigHelp()`:

```
  if (!!argv.extensions) {
    log(green('Building extension(s):'), cyan(argv.extensions));
  }
```

Reply:
Finally, remove the line
```
gulp.task('extensions', 'Build AMP Extensions', buildExtensions);
```
... from the bottom of the file, and add documentation to `gulp watch` and `gulp build` for the new `--extensions` flag.

Reply:
It might also be worth adding a `--noextensions` flag to `gulp watch` and `gulp build` that entirely skips building extensions.

Reply:
Finally, don't forget to update `DEVELOPING.md` with the new changes.